adagrams/game.py

```diff
@@ -1,11 +1,212 @@
+        selection = random.choice(list(LETTER_POOL))
+        >if selection in LTTTER_POOL, LETTER_POOL[selection] -= 1
+            >if LETTER_POOL[selection] < 1, remove it(need deepcopy of dict)
+            > ten_letters.append
+    """
+    ten_letters = [] 
+    LETTER_POOL_02 = {}
+    for key, value in LETTER_POOL.items():
+        LETTER_POOL_02[key] = value
+    
+    
+    while len(ten_letters) < 10:
+        selection = random.choice(list(LETTER_POOL_02))
+        if selection in LETTER_POOL:
+            LETTER_POOL_02[selection] -= 1
+            if LETTER_POOL_02[selection] < 1:
+                LETTER_POOL_02.pop(selection)
+            ten_letters.append(selection)
+    return ten_letters
+
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    """
+    check if the word is an anagram of some or all of the given letters in the letter bank
+
+    created a freq dict of letter_bank: letter_bank_dict
+    created a freq dict of word: word_dict
+    for letter in word_dict:
+        if letter in letter_bank_dict:
+            if word_dict[letter] < letter_bank_dict[letter]  #compare the freq of same letter
+            return True
+
+    input.upper()
+    created a freq dict of letter_bank: letter_bank_dict
+    for letter in word:
+        if letter in letter_bank_dict:
+            letter_bank_dict[letter] -= 1
+            if letter_bank_dict[letter] < 1:
+                return False
+            else: 
+                return True        
+    """
+
+    # letter_bank = draw_letters()
+    letter_bank_dict = {}
+    word = word.upper()
+    for letter in letter_bank:
+        if letter in letter_bank_dict:
+            letter_bank_dict[letter] += 1
+        else:
+            letter_bank_dict[letter] = 1
+ 
+
+    for letter in word:
+        if letter in letter_bank_dict:
+            letter_bank_dict[letter] -= 1
+            if letter_bank_dict[letter] < 0:
+                return False
+        
+        else:
+            return False
+    return True
 
 def score_word(word):
-    pass
+    """
+    input: word
+    output: total score of the word: sum of letters' point value + bonus of lenth value 8 (if lenth >= 7)
+    score= 0
+    for letter in word:
+        for key, value in score_chart:
+            if letter in value:
+                score += key
+
+    if lenth > 7 and lenth < 11:
+        score += 8
+        
+scores_chart = {"A": 1, "C": 3, "B": 3, "E": 1, "D": 2, "G": 2, 
+         "F": 4, "I": 1, "H": 4, "K": 5, "J": 8, "M": 3, 
+         "L": 1, "O": 1, "N": 1, "Q": 10, "P": 3, "S": 1, 
+         "R": 1, "U": 1, "T": 1, "W": 4, "V": 4, "Y": 4, 
+         "X": 8, "Z": 10}
+    """
+    word = word.upper()
+    score = 0
+    for letter in word:
+        score += scores_chart[letter]
+    if len(word) >= 7 and len(word)< 11:
+        # if len(word) == 7 or len(word) == 8 or len(word) == 9 or len(word) == 10:
+        # if len(word) in [7, 8, 9, 10]:
+        score +=  8
 
+    return score 
```

Comment:
We can add `upper()` right in the for loop instead of reassigning `word` to save us a line of code.

```python
    score = 0
    for letter in word.upper():
        score += scores_chart[letter]
    if len(word) >= 7 and len(word)< 11:
        # if len(word) == 7 or len(word) == 8 or len(word) == 9 or len(word) == 10:
        # if len(word) in [7, 8, 9, 10]:
        score +=  8
    return score 
```